Microsoft.Toolkit.Uwp.UI.Controls/Carousel/CarouselAutomationPeer.cs

```diff
@@ -0,0 +1,148 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Toolkit.Uwp.UI.Controls;
+using Windows.UI.Xaml.Automation.Peers;
+using Windows.UI.Xaml.Automation.Provider;
+using Windows.UI.Xaml.Controls;
+
+namespace Microsoft.Toolkit.Uwp.UI.Automation.Peers
+{
+    /// <summary>
+    /// Defines a framework element automation peer for the <see cref="Carousel"/> control.
+    /// </summary>
+    public class CarouselAutomationPeer : ItemsControlAutomationPeer, ISelectionProvider
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CarouselAutomationPeer"/> class.
+        /// </summary>
+        /// <param name="owner">
+        /// The <see cref="Carousel" /> that is associated with this <see cref="T:Windows.UI.Xaml.Automation.Peers.CarouselAutomationPeer" />.
+        /// </param>
+        public CarouselAutomationPeer(Carousel owner)
+            : base(owner)
+        {
+        }
+
+        /// <summary>Gets a value indicating whether the Microsoft UI Automation provider allows more than one child element to be selected concurrently.</summary>
+        /// <returns>True if multiple selection is allowed; otherwise, false.</returns>
+        public bool CanSelectMultiple => false;
+
+        /// <summary>Gets a value indicating whether the UI Automation provider requires at least one child element to be selected.</summary>
+        /// <returns>True if selection is required; otherwise, false.</returns>
+        public bool IsSelectionRequired => false;
+
+        private Carousel OwningCarousel
+        {
+            get
+            {
+                return Owner as Carousel;
+            }
+        }
+
+        /// <summary>Retrieves a UI Automation provider for each child element that is selected.</summary>
+        /// <returns>An array of UI Automation providers.</returns>
+        public IRawElementProviderSimple[] GetSelection()
+        {
+            CarouselItem selectedCarouselItem = OwningCarousel.GetCarouselItems().FirstOrDefault(x => x.IsSelected);
```

Comment:
QQ @jamesmcroft: Carousel has a `SelectedItem` property, so is there a reason we can't use that here vs. enumerating the collection?

Reply:
Good point, I don't see why not. I'll make the change and run the automation tooling but I can see this working.

Reply:
Sorted, that worked perfectly!